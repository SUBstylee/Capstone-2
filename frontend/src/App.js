import { Container } from "react-bootstrap";
import { Route, Routes } from "react-router-dom";

import Header from "./components/Header/Header";
import Footer from "./components/Footer/Footer";
import Home from "./pages/Home/Home";
import ProductDetails from "./pages/ProductDetails/ProductDetails";
import Cart from "./pages/Cart/Cart";
import Login from "./pages/Login/Login";
import Register from "./pages/Register/Register";
import Profile from "./pages/Profile/Profile";
import Shipping from "./pages/Shipping/Shipping";
import Payment from "./pages/Payment/Payment";
import PlaceOrder from "./pages/PlaceOrder/PlaceOrder";
import Order from "./pages/Order/Order";
import UserList from "./pages/UserList/UserList";
import UserEdit from "./pages/UserEdit/UserEdit";
const App = () => {
  return (
    <div className="App">
      <Header />
      <h1>Welcome to Totally Awesome Apparel</h1>
      <main className="py-3">
        <Container>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/products/:id" element={<ProductDetails />} />
            <Route path="/login" element={<Login />} />
            <Route path="/register" element={<Register />} />
            <Route path="/profile" element={<Profile />} />
            <Route path="/shipping" element={<Shipping />} />
            <Route path="/payment" element={<Payment />} />
            <Route path="/placeorder" element={<PlaceOrder />} />
            <Route path="/order/:id" element={<Order />} />
            <Route path="/admin/userlist" element={<UserList />} />
            <Route path="/admin/user/:id/edit" element={<UserEdit />} />
            <Route path='/cart'>
              <Route path=":id" element={<Cart />} />
              <Route path="" element={<Cart />} />
            </Route>
          </Routes>
        </Container>
      </main>
      <Footer />
    </div>

  );
};

export default App;
