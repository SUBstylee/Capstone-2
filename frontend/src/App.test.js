import { render, screen } from '@testing-library/react';
import App from './App';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from './store';

test('renders App component without crashing', () => {
  const { getByText } = render(<Provider store={store}><BrowserRouter><App /></BrowserRouter></Provider>);
  expect(getByText("Welcome to Totally Awesome Apparel")).toBeInTheDocument();
});

test("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}><BrowserRouter><App /></BrowserRouter></Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
