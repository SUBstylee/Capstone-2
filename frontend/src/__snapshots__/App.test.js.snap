// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="Header"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <a
          class="active navbar-brand"
          href="/"
        >
          <img
            alt="Totally Awesome Apparel Logo"
            class="ps-2"
            height="36"
            src="TAA-header-color-transparent.png"
          />
        </a>
        <div
          class="container"
        >
          <button
            aria-controls="basic-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="basic-navbar-nav"
          >
            <div
              class="ms-auto navbar-nav"
            >
              <a
                class="nav-link"
                data-rr-ui-event-key="/cart"
                href="/cart"
              >
                <span
                  class="fa-layers fa-fw"
                >
                  <i
                    class="fas fa-shopping-cart fa-lg"
                  />
                  <span
                    class="fa-layers-counter fa-layers-top-left"
                    style="background: rgb(252, 145, 126); border-radius: 50%; padding: 0px 6px; margin-right: 1px;"
                  >
                    0
                  </span>
                </span>
                Cart
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="/login"
                href="/login"
              >
                <i
                  class="fa-solid fa-right-to-bracket"
                />
                 Sign In
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="/register"
                href="/register"
              >
                <i
                  class="fa-solid fa-user-plus"
                />
                 Register
              </a>
              <div
                class="Search "
              >
                <form
                  class=""
                >
                  <input
                    class="Search-input"
                    name="q"
                    placeholder="Search Products..."
                    type="text"
                  />
                </form>
                <button
                  class="Search-btn"
                >
                  <i
                    class="fas fa-search"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </nav>
      <nav
        class="Header-cats navbar navbar-expand navbar-light bg-white"
      >
        <div
          class="mx-auto navbar-nav"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="/search/__Hats__"
            href="/search/__Hats__"
          >
            <i
              class="fa-solid fa-hat-cowboy"
            />
             Hats
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/search/__Shoes__"
            href="/search/__Shoes__"
          >
            <i
              class="fa-solid fa-shoe-prints"
            />
             Shoes
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/search/__Jackets__"
            href="/search/__Jackets__"
          >
            <i
              class="fa-solid fa-shirt"
            />
             Jackets
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/search/__Women__"
            href="/search/__Women__"
          >
            <i
              class="fa-solid fa-person"
            />
             Women
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/search/__Men__"
            href="/search/__Men__"
          >
            <i
              class="fa-solid fa-person-dress"
            />
             Men
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/search/__All__"
            href="/search/__All__"
          >
            <i
              class="fa-solid fa-list-ul"
            />
             All
          </a>
        </div>
      </nav>
    </header>
    <main
      class="py-3"
    >
      <div
        class="container"
      >
        <div
          class="Home"
        >
          <h1>
            Welcome to Totally Awesome Apparel!
          </h1>
          <div
            class="Toast "
            style="top: 50px; right: 25px;"
          >
            <button
              class="btn-sm btn btn-danger"
              type="button"
            >
              <i
                class="fa-solid fa-x"
              />
            </button>
            <h4>
              Not a real store!
            </h4>
            <p>
              This site was built using the MERN stack. Payments are in test mode, so you will not be charged. There are no real products here.
            </p>
          </div>
          <div
            class="Directory"
          >
            <div
              class="MenuItem"
            >
              <div
                class="background-image"
                style="background-image: url(/assets/images/shop-img/hats.png);"
              />
              <div
                class="content"
              >
                <h1
                  class="title"
                >
                  HATS
                </h1>
                <span
                  class="subtitle"
                >
                  SHOP NOW
                </span>
              </div>
            </div>
            <div
              class="MenuItem"
            >
              <div
                class="background-image"
                style="background-image: url(/assets/images/shop-img/jackets.png);"
              />
              <div
                class="content"
              >
                <h1
                  class="title"
                >
                  JACKETS
                </h1>
                <span
                  class="subtitle"
                >
                  SHOP NOW
                </span>
              </div>
            </div>
            <div
              class="MenuItem"
            >
              <div
                class="background-image"
                style="background-image: url(/assets/images/shop-img/sneakers.png);"
              />
              <div
                class="content"
              >
                <h1
                  class="title"
                >
                  SHOES
                </h1>
                <span
                  class="subtitle"
                >
                  SHOP NOW
                </span>
              </div>
            </div>
            <div
              class="MenuItem"
            >
              <div
                class="background-image"
                style="background-image: url(/assets/images/shop-img/womens.png);"
              />
              <div
                class="content"
              >
                <h1
                  class="title"
                >
                  WOMEN
                </h1>
                <span
                  class="subtitle"
                >
                  SHOP NOW
                </span>
              </div>
            </div>
            <div
              class="MenuItem"
            >
              <div
                class="background-image"
                style="background-image: url(/assets/images/shop-img/men.png);"
              />
              <div
                class="content"
              >
                <h1
                  class="title"
                >
                  MEN
                </h1>
                <span
                  class="subtitle"
                >
                  SHOP NOW
                </span>
              </div>
            </div>
            <div
              class="MenuItem"
            >
              <div
                class="background-image"
                style="background-image: url(/assets/images/shop-img/sample.png);"
              />
              <div
                class="content"
              >
                <h1
                  class="title"
                >
                  ALL
                </h1>
                <span
                  class="subtitle"
                >
                  SHOP NOW
                </span>
              </div>
            </div>
          </div>
          <div
            class="TopProducts"
          >
            <div
              class="Loader"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="Footer"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="text-center col"
          >
            Copyright Â© Totally Awesome Apparel
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="text-center col"
          >
            <a
              class="btn"
              href="https://github.com/SUBstylee"
              rel="noopener noreferrer"
              role="button"
              style="background-color: rgb(51, 51, 51); color: rgb(255, 255, 255);"
              target="_blank"
            >
              <i
                class="fab fa-github"
              />
            </a>
            <a
              class="btn"
              href="https://www.linkedin.com/in/jeremy-threlfall/"
              rel="noopener noreferrer"
              role="button"
              style="background-color: rgb(0, 130, 202); color: rgb(255, 255, 255);"
              target="_blank"
            >
              <i
                class="fab fa-linkedin-in"
              />
            </a>
            <a
              class="btn"
              href="https://line.me/R/ti/p/substylee"
              rel="noopener noreferrer"
              role="button"
              style="background-color: rgb(6, 199, 85); color: rgb(255, 255, 255);"
              target="_blank"
            >
              <i
                class="fab fa-line"
              />
            </a>
            <a
              class="btn"
              href="https://twitter.com/jjthrelfall"
              rel="noopener noreferrer"
              role="button"
              style="background-color: rgb(85, 172, 238); color: rgb(255, 255, 255);"
              target="_blank"
            >
              <i
                class="fab fa-twitter"
              />
            </a>
            <a
              class="btn"
              href="https://www.facebook.com/substylee"
              rel="noopener noreferrer"
              role="button"
              style="background-color: rgb(59, 89, 152); color: rgb(255, 255, 255);"
              target="_blank"
            >
              <i
                class="fab fa-facebook-f"
              />
            </a>
            <a
              class="btn"
              href="https://www.instagram.com/jjthrelfall/"
              rel="noopener noreferrer"
              role="button"
              style="background-color: rgb(172, 43, 172); color: rgb(255, 255, 255);"
              target="_blank"
            >
              <i
                class="fab fa-instagram"
              />
            </a>
          </div>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
